lib/new_relic/agent/configuration/default_source.rb

```diff
@@ -2212,6 +2212,73 @@ def self.enforce_fallback(allowed_values: nil, fallback: nil)
           :type => Integer,
           :allowed_from_server => false,
           :description => 'This value represents the total amount of memory available to the host (not the process), in mebibytes (1024 squared or 1,048,576 bytes).'
+        },
+        # security agent
+        :'security.agent.enabled' => {
+          :default => false,
+          :external => true,
+          :public => true,
+          :type => Boolean,
+          :allowed_from_server => false,
+          :description => 'If `true`, the security agent is loaded'
+        },
+        :'security.enabled' => {
+          :default => false,
+          :external => true,
+          :public => true,
+          :type => Boolean,
+          :allowed_from_server => false,
+          :description => 'If `true`, the security agent is activated'
+        },
+        :'security.mode' => {
+          :default => 'IAST',
+          :external => true,
+          :public => true,
+          :type => String,
+          :allowed_from_server => true,
+          :description => 'Defines the mode for the security agent to operate in. Currently only `IAST` is supported',
+          :dynamic_name => true
+        },
+        :'security.validator_service_url' => {
+          :default => 'wss://csec.nr-data.net',
+          :external => true,
+          :public => true,
+          :type => String,
+          :allowed_from_server => true,
+          :description => 'Defines the end point URL for posting security related data',
+          :dynamic_name => true
+        },
+        :'security.detection.rci.enabled' => {
+          :default => true,
+          :external => true,
+          :public => true,
+          :type => Boolean,
+          :allowed_from_server => false,
+          :description => 'If `true`, enables rci detection'
+        },
+        :'security.detection.rxss.enabled' => {
+          :default => true,
+          :external => true,
+          :public => true,
+          :type => Boolean,
+          :allowed_from_server => false,
+          :description => 'If `true`, enables rxss detection'
+        },
+        :'security.detection.deserialization.enabled' => {
+          :default => true,
+          :external => true,
+          :public => true,
+          :type => Boolean,
+          :allowed_from_server => false,
+          :description => 'If `true`, enables deserialization detection'
+        },
+        :'security.applicationinfo.port' => {
+          :default => -1,
```

Comment:
Could we replace this value with the default port for Passenger? I believe it's [3000](https://www.phusionpassenger.com/library/config/standalone/intro.html#configuring-through-command-line-options-environment-variables-and-passengerfile-json)

Reply:
Even though it's needed for Passenger, it's also used if populated by the other web servers. The `-1` seems to be a trick that allows the use of a numeric field to convey "don't use the actual number I'm giving you, use the default". I wonder if `nil` would work in the same way. No matter what, we'd need a change made on the security agent side before we can change this value here.

Reply:
We have precedence in our code to have `nil` defaults for `Integer` configurations with `utilization.logical_processors` and `utilization.total_ram_mib`. I think I would prefer that approach over `-1`.

Reply:
This option is Ruby specific, so we have some leeway. We can submit a PR to replace -1 in both code bases. I can take that or you can.

Reply:
I like how down here I realized correctly that it's Ruby specific.